### PR TITLE
Fix crash in EditorPropertyArray

### LINKED ISSUE #51298: Editor crashes when resizing array fields
### Godot version

Official 3.3.2

### System information

macOS Big Sur (11.5), 2019 MacBook Pro 16"

### Issue description

When reducing the size of an array (such as a PoolVector2Array or PoolColorArray on a Polygon2D) field in the Inspector, Godot crashes.

Screen Shot 2021-08-05 at 4 25 48 PM

Screen Shot 2021-08-05 at 4 26 05 PM

### Steps to reproduce

1. Add any node that contains an Array field (such as a Polygon2D or CollisionShape2D). (first pic in description)
2. Pick any array field and click to create the array and enter the size.
3. Click or tab out of the size field so the array is created. (second pic in description)
4. Change the size to something smaller and click or tab out of the field.
5. Godot crashes.

This happens on brand new projects with no other scenes, and on full blown projects.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #51298 (probably)
I managed to trigger the condition a few times and prevent a crash, but not sure if it prevents all crashes.

### CODE REVIEW

File: editor/editor_properties_array_dict.cpp [hunk 1]
```diff
@@ -655,6 +655,10 @@ void EditorPropertyArray::_reorder_button_up() {
 	reorder_mouse_y_delta = 0.0f;
 
 	Input::get_singleton()->set_mouse_mode(Input::MOUSE_MODE_VISIBLE);
+
+	if (!reorder_selected_button) {
+		return;
```
Comment (LightningAA) [inline editor/editor_properties_array_dict.cpp:660]: Hmm... Not sure about this. The only place `reorder_selected_button` is set is in `_reorder_button_down` and `_reorder_button_up`, so either `_reorder_button_up` is getting called twice in a row, or the button somehow becomes null after `_reorder_button_down` (but on `_reorder_button_down` the mouse gets captured so there's not much the user can do). If it fixes the crashing, that's something I guess, but I might try debugging this later see if I can figure anything out.

Comment (LightningAA) [reply editor/editor_properties_array_dict.cpp:660]: Also, I can reproduce the crash in 3.3, where this code doesn't exist.

Comment (KoBeWi) [reply editor/editor_properties_array_dict.cpp:660]: Well it DOES fix some crash, so maybe mine is unrelated to the original issue.

**Comment by KoBeWi** [comment]
IIRC the Array inspector had some changes since this PR was opened and the issue is likely no longer valid. I can't seem to reproduce it anymore.

**Review by groud — APPROVED** [review]

**Comment by groud** [comment]
It looks good to me. I am not sure if there's a reason why `reorder_selected_button = Object::cast_to<Button>(reorder_selected_element_hbox->get_child(0));` would fail or not (maybe the child could be missing ?). But I think the check is welcome anyway.

**Comment by akien-mga** [comment]
I can still reproduce the crash using those steps in current 3.x: https://github.com/godotengine/godot/issues/51298#issuecomment-894078651

I can't reproduce in 4.0, possibly because I can't validate input in the `size` field with Tab due to another bug :P

**Comment by KoBeWi** [comment]
I think this might be only relevant for 3.x now.

**Comment by akien-mga** [comment]
I'd suggest changing this to an `ERR_FAIL_COND` as this is not an expected scenario. So it should fix the crash but still make some noise so we're aware that there is still a race condition. Could be merged for master and cherry-picked even if the bug doesn't seem reproducible in master.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.6.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.2.

**Pull request merged**